activerecord/test/cases/persistence_test.rb

```diff
@@ -120,6 +120,15 @@ def test_increment_attribute_by
     assert_equal 59, accounts(:signals37, :reload).credit_limit
   end
 
+  def test_increment_concurency
+    a1 = accounts(:signals37)
+    initial_credit = a1.credit_limit
+    a2 = Account.find(accounts(:signals37).id)
+    a1.increment!(:credit_limit)
+    a2.increment!(:credit_limit)
+    assert_equal initial_credit + 2, a1.reload.credit_limit
+  end
+
```

Comment:
"test increment concurrency" needs to be more precise so future readers of this code understand the guarantee we're making. This is testing the *I* in ACID, that `#increment!` isn't vulnerable to stale reads.

Does this single test really cover all the code changes above? It's difficult to determine but seems unlikely.